rs/libmoq/src/state.rs

```diff
@@ -47,7 +47,12 @@ impl DerefMut for StateGuard {
 impl State {
 	pub fn lock() -> StateGuard {
 		let runtime = RUNTIME.enter();
-		let state = STATE.lock().unwrap();
+		let state = STATE.lock().unwrap_or_else(|poisoned| {
+			// If the mutex is poisoned due to a previous panic, we still want to continue
+			// Recover the state from the poisoned mutex
+			tracing::warn!("Mutex was poisoned, recovering from previous panic");
+			poisoned.into_inner()
+		});
```

Comment:
Too much vibe coding. Everything should crash if we panic. Warning and continuing anyway is just bad.